react_app/src/hooks/useSearches.tsx

```diff
@@ -53,34 +54,37 @@ export function useSearch() {
   const [searchResults, setSearchResults] = useState<Record<Search['id'], Search['result']>>({});
 
   // Poll for search status changes
-  const shouldPoll = searchIds.some(id => searchStatus[id] !== "completed");
+  const pendingSerialized = JSON.stringify(
+    searchIds.filter(id => searchStatus[id] !== "completed")
+  );
```

Comment:
Proper `useEffect` dependencies must be `==` comparable for identity. Serializing to a JSON string works well here (and below for `completedSerialized`) and is more readable IMO than then more complicated `useRef` hack.